integration-test/integration-test.ps1

```diff
@@ -0,0 +1,287 @@
+#>
+   Write-Host $sJsonLocation
+
+   [PSCustomObject]$json = [PSCustomObject]((Get-Content -Raw $sJsonLocation) -replace '(?m)(?<=^([^"]|"[^"]*")*)//.*' -replace '(?ms)/\*.*?\*/' | Out-String | ConvertFrom-Json)
+
+   foreach ($jsCmd in $json) {
+    $validateOutput = $null
+
+    if($prop = $jsCmd.PSObject.Properties.Item('expectOutput')) {
+        write-host ("Check output for :" + $prop.Value)
+        $validateOutput = $jsCmd.expectOutput
+    }
+    else {
+        write-host "No output found"
+    }
+
+    #Write-Host $jsCmd.commands + $jsCmd.arguments + $validateOutput -ForegroundColor Cyan
+    if ($prop = $jsCmd.PSObject.Properties.Item('waitSeconds')) {
+        Write-Host "Waiting " $jsCmd.waitSeconds " seconds to run this command." -ForegroundColor DarkYellow
+        Start-Sleep -Seconds $jsCmd.waitSeconds
+    }
+    Run_Command -sCmd $jsCmd.commands -argument $jsCmd.arguments -valOutput $jsCmd.expectOutput
+
+   }
+
+}
+
+Function Run_Command($sCmd, $argument, $valOutput) {
+<#
+.SYNOPSIS
+	Run each command as defined.
+.DESCRIPTION
+	This PowerShell function execute command as defined in json file and output to the console.
+#>
+ Write-Host ("command: " + $sCmd)
+ Write-Host ("arg: " + $argument)
+ Write-Host ("Output: " + $valOutput)
+
+ #$global:blnLinux = ($env:OS -eq "" -or $env:OS -eq $null)
+ if ($global:blnLinux) {
+    $sCmd = "java -jar " + $global:FCLI_MODULE + " " + $sCmd + " " + $argument
+ }else{
+    #$sCmd = "./" + $global:FCLI_MODULE + " " + $sCmd + " " + $argument
+    $sCmd = $global:FCLI_MODULE + " " + $sCmd + " " + $argument
+ }
+
+ Write-Host ("Running Command: $($sCmd)") -ForegroundColor Cyan
+
+ $retVal = Invoke-Expression $sCmd -ErrorAction Continue | Out-String
+ Write-Host "Testing results returned:`n" $retVal
+
+ if ($valOutput -ne $null -and $retVal -ne $null)
+ {
+
+    $found = $retVal -match $valOutput
+
+    if ($found)
+    {
+        Write-Host "Test Result matched with Output parameter, PASSED" -ForegroundColor Green
+    } else {
+        Write-Host "Test Result did not matched with Output parameter, FAILS..." -ForegroundColor Red
+    }
+ }
+ if ($valOutput -eq $null -and $retVal -ne $null)
+ {
+    Write-Host "TEST PASSED..." -ForegroundColor Green
+ } 
+ if ($valOutput -eq $null -and $retVal -eq $null)
+ {
+    Write-Host "TEST FAILED..." -ForegroundColor Red
+ } 
+}
+
+#
+# Main 
+#
+Set-LocalEnvironmentVariable "FCLI_TEST_MODULES" "fod, ssc, scsast"
+$global:FCLI_MODULE="fcli-beta.exe"
+$global:blnLinux = ($env:OS -eq "" -or $env:OS -eq $null)
+$MODULE_INPUT_STR="Enter name and path of the command line tool along with extension[fcli.exe]"
+if ($global:blnLinux) {
+    $MODULE_INPUT_STR="Enter name and path of the command line tool along with extension[fcli.jar]"
+    $global:FCLI_MODULE="fcli-beta.jar"
+}
+
+$ErrorActionPreference="SilentlyContinue"
+Stop-Transcript | out-null
+$ErrorActionPreference = "Continue"
+Start-Transcript -path .\fcli-$(get-date -f yyyy-MM-dd).log -append

```

Comment:
Start-Transcript is no log for logging results, as it doesn't log the errors from fcli just the commands.

We need to be able to produce a consolidated report of tests pass/fail